app/services/catalog/notify.rb

```diff
@@ -0,0 +1,28 @@
+module Catalog
+  class Notify
+    ACCEPTABLE_NOTIFICATION_CLASSES = %w[OrderItem ApprovalRequest].freeze
+
+    attr_reader :notification_object
+
+    def initialize(klass, id, payload)
+      raise Catalog::InvalidNotificationClass unless ACCEPTABLE_NOTIFICATION_CLASSES.include?(klass.camelcase)
+
+      @notification_object = klass.camelcase.constantize.find(id)
+      @payload = payload
+    end
+
+    def process
+      @notification_object.update(:state => @payload["decision"])
```

Comment:
@eclarizio Here are the events.
```
    :event  => 'request_started',
    :payload  => {
      :request_id => '3'
    }

    :event  => 'request_finished',
    :payload  => {
      :request_id => '3',
      :decision =>"approved", 
      :reason=>"System approved"
    }

    :event  => 'request_finished',
    :payload  => {
      :request_id => '1',
      :decision =>"denied", 
      :reason=>"System Denied"
    }
```
For the `request_started` event there is no `decision` or `reason`, so this `state` update would have to change.